### PR TITLE
Add robots metadata for customized 404 page

### PR DESCRIPTION
**Pull request by 928PJY**
[AB#190733](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/190733)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/5663)

### CODE REVIEW

File: src/docfx/build/page/BuildPage.cs [hunk 1]
```diff
@@ -187,6 +187,11 @@ private static (List<Error>, SystemMetadata) CreateSystemMetadata(Context contex
             systemMetadata.SearchProduct = context.Config.Product;
             systemMetadata.SearchDocsetName = context.Config.Name;
 
+            if (IsCustomized404Page(file))
+            {
+                systemMetadata.Robots = "NOINDEX, NOFOLLOW";
```
Comment (yufeih) [inline src/docfx/build/page/BuildPage.cs]: Do we have to add the metadata pragmatically? The true page https://github.com/MicrosoftDocs/DocsRoot/blob/a5d336d517b20b65af10d57ce831818767b22adb/DocsCoreContent/404.md already has configured `noindex, nofollow` as metadata.

Comment (928PJY) [reply src/docfx/build/page/BuildPage.cs]: At least, it is feature of v2, and there are some 404 files([bagie/docs](https://bagie.visualstudio.com/DefaultCollection/tools/_git/docs?path=%2FCTAS%2F404.md),  [MicrosoftDocs/msrd-dev-center-pr](https://github.com/MicrosoftDocs/msrd-dev-center-pr/blob/master/articles/404.md)) which doesn't have this metadata

Comment (yufeih) [reply src/docfx/build/page/BuildPage.cs]: I see. How about moving line 51 down here so the check, report, action happens at the same place?

File: src/docfx/build/page/BuildPage.cs [hunk 2]
```diff
@@ -377,5 +382,10 @@ private static string CreateContent(Context context, Document file, JObject page
             ValidateBookmarks(context, file, htmlDom);
             return CreateHtmlContent(context, htmlDom);
         }
+
+        private static bool IsCustomized404Page(Document file)
+        {
+            return Path.GetFileName(file.FilePath.Path).Equals("404.md", PathUtility.PathComparison);
```
Comment (yufeih) [inline src/docfx/build/page/BuildPage.cs]: Keep the original behavior, cause a user could write a 404.yml and published to 404

Comment (928PJY) [reply src/docfx/build/page/BuildPage.cs]: For v2, only the `404.md` is treated as a customized 404 page. do we want to keep the behavior?

Comment (yufeih) [reply src/docfx/build/page/BuildPage.cs]: That was a bug (hole) in v2 🤣 ,

File: src/docfx/build/page/BuildPage.cs (hunk 1 above)
Comment (928PJY) [reply src/docfx/build/page/BuildPage.cs]: Yeah... I have thought about that, but do you think that is a little bit wired to report this warning in a function called `CreateSystemMetadata`. I can do the move if you think that would be better~

File: src/docfx/build/page/BuildPage.cs (hunk 2 above)
Comment (928PJY) [reply src/docfx/build/page/BuildPage.cs]: Got you, the `404.md` and `404.yml` will both be consumed as 404 file in DHS, I will do the change

File: src/docfx/build/page/BuildPage.cs (hunk 1 above)
Comment (yufeih) [reply src/docfx/build/page/BuildPage.cs]: Yes, it is better, `CreateSystemMetadata` already reports tons of warnings.

File: src/docfx/build/page/BuildPage.cs [hunk 3]
```diff
@@ -187,6 +181,12 @@ private static (List<Error>, SystemMetadata) CreateSystemMetadata(Context contex
             systemMetadata.SearchProduct = context.Config.Product;
             systemMetadata.SearchDocsetName = context.Config.Name;
 
+            if (IsCustomized404Page(file))
```
Comment (yufeih) [inline src/docfx/build/page/BuildPage.cs]: Move these before line 151 so make dry run work

**Pull request merged**